spec/models/user_spec.rb

```diff
@@ -38,4 +38,19 @@
       expect(subject.password_required?).to be false
     end
   end
+
+  describe "#admin?" do
+    it "is expected to be true when the user has an admin profile" do
+      user = create(:admin_profile).user
```

Comment:
This logic looks circular and a bit confusing. Why not just create the user first i.e. user = create(:user) and then create the profile with that user i.e create(:admin_profile, user: user)

Reply:
Laziness mostly... I might create a helper method that does that, because we do this often enough that it would be good to keep as a oneliner. Same for lead provider users

Reply:
Yes, I think you should fix this. In addition to what I said earlier , I'm sure it creates 2 admins when it runs and the extra admin is unnecessary.

Reply:
Could you explain how it creates two please?

Reply:
How does that look?